### ISSUE TITLE
Use % to also trigger Emmet: Go To Matching Tag

### ISSUE THREAD

**Issue by trusktr** [VSCodeVim/Vim#3872]
It would be super sweet if the `%` key also triggered `Emmet: Go To Matching Tag` (built-in command that ships with VS Code) when on HTML tags.

**Comment by mikowl** [comment]
One thing you can do for now is map "editor.emmet.action.matchTag" to a key.  I've mapped <leader>% to run editor.emmet.action.matchTag.
```
"vim.normalModeKeyBindingsNonRecursive": [
    {
      "before": ["<leader>", "%"],
      "commands": [
        {
          "command": "editor.emmet.action.matchTag"
        }
      ]
    }
]
```

**Comment by trusktr** [comment]
That's cool. Maybe I can make my own function, and have `%` trigger my function, then in my function I can conditionally call one or the other? Can you guide me the direction for this? It's very easy in Atom (which seems a lot easier to extend).

**Comment by uloco** [comment]
Unfortunately emmet go to matching tag does not work in react native. Just FYI.

**Comment by trusktr** [comment]
This works out of the box with the [`Neo Vim`](https://marketplace.visualstudio.com/items?itemName=asvetliakov.vscode-neovim) VS Code extension (real Neovim controls the VS Code text buffers), including in JSX expressions (f.e. React Native @uloco).

![neo-vim-go-to-matching-JSX](https://user-images.githubusercontent.com/297678/102924855-30fa2a00-4447-11eb-9e04-52cdfab64c98.gif)

**Comment by gaoqiangks** [comment]
It's 2023 now. Is this still open?

**Comment by sdykae** [comment]
should work on this but the config works fine ;D

> One thing you can do for now is map "editor.emmet.action.matchTag" to a key. I've mapped <leader> % to run editor.emmet.action.matchTag.
> 
> ```
> "vim.normalModeKeyBindingsNonRecursive": [
>     {
>       "before": ["<leader>", "%"],
>       "commands": [
>         {
>           "command": "editor.emmet.action.matchTag"
>         }
>       ]
>     }
> ]
> ```

**Comment by HabibMollah** [comment]
> it is not for visual mode even config this:
> 
> ```
>   "vim.visualModeKeyBindingsNonRecursive": [
>     {
>       "before": ["<leader>", "%"],
>       "commands": [
>         {
>           "command": "editor.emmet.action.matchTag"
>         }
>       ]
>     }
>   ]
> ```

Does anyone know how to configure it for visual mode properly to select from open tag to closing tag?

**Comment by LukaErnestini** [comment]
@HabibMollah Isn't `vat` good for selecting whole tag? (or `vit` for just the content inside)

**Comment by HabibMollah** [comment]
> @HabibMollah Isn't `vat` good for selecting whole tag? (or `vit` for just the content inside)

@LukaErnestini Yes it is! Found out about this motion few days later after I posted the comment above.

**Comment by luga97** [comment]
> @HabibMollah Isn't `vat` good for selecting whole tag? (or `vit` for just the content inside)

for function/block you can use `va{` or `vi{` so useful

**Comment by chuanqisun** [comment]
I'm tinkering on a workaround, how about fully overwrite vim's handling of % key?
```json
"vim.normalModeKeyBindingsNonRecursive": [
  {
    "before": ["%"],
    "commands": [
      {
        "command": "editor.emmet.action.matchTag"
      },
      {
        "command": "editor.action.jumpToBracket"
      }
    ]
  }
],
```
